extension/js/common/core/crypto/pgp/openpgp-key.ts

```diff
@@ -505,40 +498,28 @@ export class OpenPGPKey {
     const verifyRes: VerifyRes = {
       match: null, // eslint-disable-line no-null/no-null
       signerLongids: [],
-      signerFingerprints: [],
       suppliedLongids: validKeys.map(x => x.allIds.map(fp => OpenPGPKey.fingerprintToLongid(fp))).reduce((a, b) => a.concat(b), []),
     };
-    const opgpKeys = validKeys.map(x => OpenPGPKey.extractExternalLibraryObjFromKey(x));
+    const opgpKeys = await Promise.all(validKeys.map(x => OpenPGPKey.extractExternalLibraryObjFromKey(x)));
     // todo: expired?
     try {
-      // this is here to ensure execution order when 1) verify, 2) read data, 3) processing signatures
-      // Else it will hang trying to read a stream: https://github.com/openpgpjs/openpgpjs/issues/916#issuecomment-510620625
-      const verifications = await msg.verify(opgpKeys); // first step
-      const stream = msg instanceof opgp.message.Message ? msg.getLiteralData() : msg.getText();
-      if (stream) {
+      const signerLongids = msg.getSigningKeyIDs().map(kid => OpenPGPKey.bytesToLongid(kid.bytes));
+      const text = msg instanceof opgp.CleartextMessage ? msg.getText() : msg.getLiteralData(); // todo: is this important?
```

Comment:
On v4 it was, at least. `getText` was for cleartext signed data, `getLiteralData` was for signed encrypted data :shrug: